mllib/src/main/scala/org/apache/spark/mllib/api/python/PythonMLLibAPI.scala

```diff
@@ -1142,12 +1141,21 @@ private[python] class PythonMLLibAPI extends Serializable {
     new RowMatrix(rows.rdd, numRows, numCols)
   }
 
+  def createRowMatrix(df: DataFrame, numRows: Long, numCols: Int): RowMatrix = {
+    require(df.schema.length == 1 && df.schema.head.dataType.getClass == classOf[VectorUDT],
+      "DataFrame must have a single vector type column")
+    new RowMatrix(df.rdd.map { case Row(vector: Vector) => vector }, numRows, numCols)
```

Comment:
This doesn't save much. In this case you can work out the number of rows and columns for the caller no? I get that this is to parallel the method below, which also doesn't do that. It should be possible to at least work out the number of rows.

Reply:
I'm not sure I follow what you mean by "doesn't save much". Could you elaborate?

From Python, you don't need to specify the number of rows / columns. They're 0 by default, which will cause `RowMatrix` to infer the dimensions.

Reply:
OK the number of rows issue seems fine then. I see, the point is that this is already simple in Scala but not in Python. I think it could be reasonable then.